ComponentKitTests/CKComponentDelegateAttributeTests.mm

```diff
@@ -110,6 +110,38 @@ - (void)testProxiedEventsProxy
   XCTAssertFalse(hierarchy.receivedScroll, @"Should not have triggered on old hierarchy either.");
 }
 
+- (void)testProxiedEventsProxyThrowsAssertionButDoesNotCrashWhenUnmounted
+{
+  CKDetectScrollComponent *hierarchy =
+  [CKDetectScrollComponent
+   newWithComponent:[CKComponent
+                     newWithView:{[UIScrollView class],
+                       {CKComponentDelegateAttribute(@selector(setDelegate:), {
+                         @selector(scrollViewDidScroll:),
+                       })}}
+                     size:{}]];
+  
+  
+  CKComponentLayout layout = [hierarchy layoutThatFits:{} parentSize:{NAN, NAN}];
+  
+  UIView *container = [UIView new];
+  NSSet *mounted = CKMountComponentLayout(layout, container, nil, nil);
+  
+  XCTAssertFalse(hierarchy.receivedScroll, @"Should not have triggered yet");
+  
+  UIScrollView *scroll = findScrollView(container);
+  
+  CKUnmountComponents(mounted);
+  
+  XCTAssertThrows(scroll.contentOffset = CGPointMake(0, 100));
```

Comment:
Before this update, this line would throw, but then hard-crash. Now it just throws.